### PR TITLE
generate color translation tables

### LINKED ISSUE #920: add Crispy's any-to-red/green/blue color translation tables
for status bar numbers and probably the menu (but not anything affecting in- game rendering)

### PR DESCRIPTION
**Pull request by fabiangreffrath**
Fixes #920

### CODE REVIEW

File: src/v_trans.c [hunk 1]
```diff
@@ -204,7 +204,7 @@ byte V_Colorize (byte *playpal, int cr, byte source)
         }
         else if (cr == CR_BLUE || cr == CR_BLUE2)
         {
-            hsv.x = (float) 240./360.;
+            hsv.x = (float) (240./360.);
```
Comment (rfomin) [inline src/v_trans.c:207]: Alternative fix: `hsv.x = 240.f/360.f;`
Why not use `double` in v_trans.c? By default, all calculations are performed in double precision.

Comment (fabiangreffrath) [reply src/v_trans.c:207]: Hm, not sure, I have taken this code from somebody else and would just leave it untouched as it "just worked". 😁

File: src/m_menu.c [hunk 2]
```diff
@@ -3303,7 +3303,7 @@ setup_menu_t stat_settings1[] =  // Status Bar and HUD Settings screen
 {
   {"STATUS BAR"        ,S_SKIP|S_TITLE,m_null,M_X,M_Y},
 
-  {"USE RED NUMBERS"   ,S_YESNO|S_COSMETIC,m_null,M_X,M_Y+stat1_rednum*M_SPC,   {"sts_always_red"}},
+  {"USE STANDARD COLORS",S_YESNO|S_COSMETIC,m_null,M_X,M_Y+stat1_rednum*M_SPC,   {"sts_always_red"}},
```
Comment (fabiangreffrath) [inline src/m_menu.c:3306]: Not sure about this wording, as so often.

File: src/m_menu.c [hunk 3]
```diff
@@ -2608,7 +2608,7 @@ void M_DrawDefVerify()
   if (whichSkull) // blink the text
     {
       strcpy(menu_buffer,"Reset to defaults? (Y or N)");
-      M_DrawMenuString(VERIFYBOXXORG+8-WIDESCREENDELTA,VERIFYBOXYORG+8,CR_RED);
+      M_DrawMenuString(VERIFYBOXXORG+8-WIDESCREENDELTA,VERIFYBOXYORG+8,CR_NONE);
```
Comment (fabiangreffrath) [inline src/m_menu.c:2611]: Should actual warnings remain red?

Comment (rfomin) [reply src/m_menu.c:2611]: Yes, I think so. In the PWADs I've tested, red looks more right. Instructions too?

**Comment by rfomin** [comment]
In general, I really like this PR, thank you.
Found some PWADs with weird/hard to read fonts/colors:
PL2.wad
woof0005
[Avactor.wad](https://www.doomworld.com/forum/topic/101647-avactor-last-update-is-out/)
woof0003
[Breathless](https://www.doomworld.com/idgames/levels/doom2/Ports/a-c/breathlessv2)
woof0004
I don't know if we should do anything about it.

**Comment by fabiangreffrath** [comment]
Does it improve if you run with `-tranmap`?

**Comment by fabiangreffrath** [comment]
Even the IWAD fonts won't be red with the modified palette.

**Comment by fabiangreffrath** [comment]
PL2.wad looks like the DOS font. 🙄  Any idea what we can do with palette without a proper orange/green/gold range? I wouldn't want to switch to a different palette whenever the menu is opened, and we would also lose the ability to have translucent menu background.

**Comment by rfomin** [comment]
> I wouldn't want to switch to a different palette whenever the menu is opened, and we would also lose the ability to have translucent menu background.

The idea is to introduce a "readability mode" with a forced background, font and palette from IWAD. Or even our own font with a free license. But perhaps this idea is out of scope for conservative port.

**Comment by fabiangreffrath** [comment]
Oh, that's really far out.

**Review by rfomin — APPROVED** [review]
LGTM

**Pull request merged**